src/masternodes/vaulthistory.h

```diff
@@ -55,14 +55,62 @@ struct VaultHistoryValue {
     }
 };
 
+struct VaultStateKey {
+    uint256 vaultID;
+    uint32_t blockHeight;
```

Comment:
Could you compare to `AuctionHistoryKey` where height is first? It's faster to navigate through records when height is specified. I plan to rework account history to match auction one.

Reply:
I do not think that height will often be specified when accessing vault records and with height first won't we have to return `true` when `isMatchVault` is `false` in `ForEachVaultHistory`, so we will end up looking through all records instead of just the ones relating to the vault specific in `getvaulthistory`?

https://github.com/DeFiCh/ain/blob/2dc3439a3248549762e192be8ff4ad8e0ab04728/src/masternodes/rpc_vault.cpp#L1322